File: src/app/conjugation/forms/potential-negative.ts
import {Conjugation, Word, WordType} from "../conjugation";
import {PotentialAffirmative} from "./potential-affirmative";
import {NonPastShortNegative} from "./non-past-short-negative";

export class PotentialNegative implements Conjugation {
    getConjugation(word: Word): (Word | undefined) {

        const originalType = word.wordType;
        const potential = new PotentialAffirmative().getConjugation(word);
        if (potential === undefined) {
            return undefined;
        }

        return new NonPastShortNegative().getConjugation(potential.changeType(WordType.IchidanVerb))?.changeType(originalType)
    }

    getTitle(): string {
        return "Potential, negative"
    }

}

File: backend/app/conjugation/forms/potential_negative.py
from ..core import Conjugation, Word, WordType
from .non_past_short_negative import NonPastShortNegative
from .potential_affirmative import PotentialAffirmative


class PotentialNegative(Conjugation):

    title = 'Potential, negative'
    settings_title = 'Negative'

    def conjugate(self, word: Word) -> Word | None:
        original_type = word.word_type
        potential = PotentialAffirmative().conjugate(word)
        if potential is None:
            return None

        negative = NonPastShortNegative().conjugate(
            potential.change_type(WordType.ICHIDAN_VERB)
        )
        if negative is None:
            return None

        return negative.change_type(original_type)
